lib/src/components/NxDropdownMenu/NxDropdownMenu.tsx

```diff
@@ -4,28 +4,108 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React, { useLayoutEffect, forwardRef } from 'react';
+import React, { useLayoutEffect, forwardRef, useState, useEffect, useRef, KeyboardEventHandler } from 'react';
 import classnames from 'classnames';
+import useMergedRef from '@react-hook/merged-ref';
 
 import { Props, propTypes } from './types';
 
 import './NxDropdownMenu.scss';
 
 export { Props };
 
+const FOCUSABLE_MENU_ITEMS_SELECTOR = 'a:not([disabled]):not(.disabled), '
++ 'button:not([disabled]):not(.disabled), '
++ 'input:not([disabled]):not(.disabled)';
+
 /**
  * This component is not currently intended for public export. It is a helper for NxDropdown and NxSegmentedButton
  * so they can reset focus when they close
  */
 /* eslint-disable-next-line react/prop-types */
 const NxDropdownMenu = forwardRef<HTMLDivElement, Props>(function NxDropdownMenu(props, ref) {
-  const { onClosing, className: classNameProp, ...attrs } = props,
-      className = classnames('nx-dropdown-menu', classNameProp);
+  const { onClosing, className: classNameProp, children, disableMenuKeyNav, ...attrs } = props;
+
+  const menuRef = useRef<HTMLDivElement>(null);
+  const mergedRef = useMergedRef(menuRef, ref);
+
+  const [focusedMenuItemIndex, setFocusedMenuItemIndex] = useState<number>(0);
 
   // onClosing must execute when this element is being removed but BEFORE it actually gets removed from the DOM
   useLayoutEffect(() => onClosing, []);
 
-  return <div ref={ref} { ...{ className, ...attrs } } />;
+  useEffect(() => {
+    if (!disableMenuKeyNav && menuRef.current) {
+      menuRef.current.focus();
+    }
+  }, [disableMenuKeyNav]);
+
+  function setFocusToMenuItem(direction: -1 | 1) {
```

Comment:
Both `NxSearchDropdown` and `NxComboxbox` have also had to implement logic to do what this is doing, which they did with a function called `adjustBtnFocus`.  As this is the third time we've needed similar logic, I think it probably makes sense to see if we can extract that out into a utility and reuse it here too.